build/azure-pipelines/build-common.yml

```diff
@@ -45,7 +45,7 @@ steps:
   displayName: 'Go: build sqlcmd'
   inputs:
     command: 'build'
-    arguments: '-o $(Build.BinariesDirectory)'
+    arguments: '-o $(Build.BinariesDirectory) -ldflags="-X main.version=$(Build.BuildNumber)"'

```

Comment:
Look at build\azure-pipelines\build-product.yml

```yml
      steps:
      - task: PowerShell@2
        displayName: Set last tag to variable
        inputs:
          targetType: 'inline'
          script: |
            $VERSION_TAG = git describe --tags (git rev-list --tags --max-count=1)
            Write-Host("##vso[task.setvariable variable=VERSION_TAG]$VERSION_TAG")
            Write-Host($VERSION_TAG)
```

This should also set the build number for the pipeline for the build to pickup